src/test/scala/no/gdl/bookapi/service/ReadServiceTest.scala

```diff
@@ -0,0 +1,105 @@
+  }
+
+  test("that editorsPickForLanguage returns empty list if no ids defined in editorspick") {
+    val editorsPick = EditorsPick(Some(1), Some(1), "nob", Seq())
+
+    when(editorsPickRepository.forLanguage("nob")).thenReturn(Some(editorsPick))
+
+    readService.editorsPickForLanguage("nob") should equal (Seq())
+  }
+
+  test("that listAvailableLanguages returns languages sorted by name") {
+    when(translationRepository.allAvailableLanguages()).thenReturn(Seq(TestData.LanguageCodeEnglish, TestData.LanguageCodeAmharic))
+    when(converterService.toApiLanguage(TestData.LanguageCodeAmharic)).thenReturn(TestData.Api.amharic)
+    when(converterService.toApiLanguage(TestData.LanguageCodeEnglish)).thenReturn(TestData.Api.english)
+
+    readService.listAvailableLanguages should equal (Seq(TestData.Api.amharic, TestData.Api.english))
+  }
+
+  test("that similarTo returns empty searchresult when given id does not exist") {
+    when(translationRepository.forBookIdAndLanguage(1, "nob")).thenReturn(None)
+    when(translationRepository.languagesFor(1)).thenReturn(Seq())
+    when(bookRepository.withId(1)).thenReturn(None)
+    when(converterService.toApiBook(None, Seq(), None)).thenReturn(None)
+    when(converterService.toApiLanguage("nob")).thenReturn(TestData.Api.norwegian_bokmal)
+
+    val searchResult = readService.similarTo(1, "nob", 10, 1)
+    searchResult.results should equal (Seq())
+    searchResult.language should equal (TestData.Api.norwegian_bokmal)
+    searchResult.page should equal (1)
+    searchResult.pageSize should equal (10)
+    searchResult.totalCount should equal (0)
+  }
+
+  test("that similarTo filters out the given id from similar books") {
+    val page = 1
+    val pageSize = 10
+
+    // For getting the book to find similar for
+    when(translationRepository.forBookIdAndLanguage(1, "nob")).thenReturn(None)
+    when(translationRepository.languagesFor(1)).thenReturn(Seq())
+    when(bookRepository.withId(1)).thenReturn(None)
+    when(converterService.toApiLanguage("nob")).thenReturn(TestData.Api.norwegian_bokmal)
+    when(converterService.toApiBook(None, Seq(), None)).thenReturn(Some(TestData.Api.DefaultBook))
+
+    // For getting book with id 2
+    val book2 = TestData.Domain.DefaultBook.copy(id = Some(2))
+    val translation2 = TestData.Domain.DefaultTranslation.copy(id = Some(2))
+    val similarBook = TestData.Api.DefaultBook.copy(id = 2)
+
+    when(translationRepository.forBookIdAndLanguage(2, "nob")).thenReturn(Some(translation2))
+    when(translationRepository.languagesFor(2)).thenReturn(Seq("nob"))
+    when(bookRepository.withId(2)).thenReturn(Some(book2))
+    when(converterService.toApiLanguage("nob")).thenReturn(TestData.Api.norwegian_bokmal)
+    when(converterService.toApiBook(Some(translation2), Seq("nob"), Some(book2))).thenReturn(Some(similarBook))
+
+    val expectedSearchResult = SearchResult[Long](2, page, pageSize, "nob", Seq(TestData.Api.DefaultBook.id, 2))
```

Comment:
Naming: Is this really "expected" results, when we use it as a mock return value instead of an assertion?